### PR TITLE
Add DeleteNamespace integration test

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Add `DeleteNamespace` integration test.

**Why?**
To cover basic delete namespace functionality.

**How did you test it?**
Run new tests.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/log/zap_logger.go [hunk 1]
```diff
@@ -56,7 +56,8 @@ var _ Logger = (*zapLogger)(nil)
 // NewTestLogger returns a logger at debug level and log into STDERR
 func NewTestLogger() *zapLogger {
 	return NewZapLogger(BuildZapLogger(Config{
-		Level: "debug",
+		// Uncomment next line of you need debug level logging in tests.
```
Comment (dnr) [inline common/log/zap_logger.go:59]:
```suggestion
		// Uncomment next line if you need debug level logging in tests.
```

File: host/namespace_delete_test.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+	}
+)
+
+func TestNamespaceSuite(t *testing.T) {
+	flag.Parse()
+	suite.Run(t, &namespaceTestSuite{})
+}
+
+func dynamicConfig() map[dynamicconfig.Key]interface{} {
+	dc := make(map[dynamicconfig.Key]interface{})
```
Comment (dnr) [inline host/namespace_delete_test.go:51]: use map literal syntax?

File: host/test_cluster.go [hunk 3]
```diff
@@ -215,6 +215,7 @@ func NewCluster(options *TestClusterConfig, logger log.Logger) (*TestCluster, er
 		WorkerConfig:                     options.WorkerConfig,
 		MockAdminClient:                  options.MockAdminClient,
 		NamespaceReplicationTaskExecutor: namespace.NewReplicationTaskExecutor(options.ClusterMetadata.CurrentClusterName, testBase.MetadataManager, logger),
+		DynamicConfigOverrides:           options.DynamicConfigOverrides,
```
Comment (dnr) [inline host/test_cluster.go:218]: Ah, I noticed this too in #3199

File: host/namespace_delete_test.go (hunk 2 above)
Comment (alexshtin) [reply host/namespace_delete_test.go]: Of course!

File: host/namespace_delete_test.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+		Namespace: "ns_name_san_diego",
+	})
+	s.NoError(err)
+	nsID := descResp.GetNamespaceInfo().GetId()
+
+	delResp, err := s.operatorClient.DeleteNamespace(ctx, &operatorservice.DeleteNamespaceRequest{
+		Namespace: "ns_name_san_diego",
+	})
+	s.NoError(err)
+	s.Equal("ns_name_san_diego-deleted-"+nsID[:5], delResp.GetDeletedNamespace())
```
Comment (dnr) [inline host/namespace_delete_test.go:139]: :slightly_frowning_face: to duplicating the renaming logic, but not a big deal

File: host/namespace_delete_test.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+	s.NoError(err)
+	s.Equal("ns_name_los_angeles-deleted-"+nsID[:5], delResp.GetDeletedNamespace())
+
+	descResp2, err := s.frontendClient.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+		Id: nsID,
+	})
+	s.NoError(err)
+	s.Equal(enumspb.NAMESPACE_STATE_DELETED, descResp2.GetNamespaceInfo().GetState())
+
+	namespaceExistsOp := func() error {
```
Comment (dnr) [inline host/namespace_delete_test.go:223]: maybe also check that DescribeWorkflowExecution returns namespace not found?

**Review by dnr — APPROVED** [review]

File: host/namespace_delete_test.go (hunk 4 above)
Comment (alexshtin) [reply host/namespace_delete_test.go:139]: There is no workaround for this. There is no way to create namespace with specific Id.

File: host/namespace_delete_test.go (hunk 5 above)
Comment (alexshtin) [reply host/namespace_delete_test.go:223]: This is actually good idea!

**Pull request merged**